Lenses/Example/src/main.js

```diff
@@ -131,14 +131,46 @@ function formatDateFields(object) {
  * This function modifies the DOM by passing the data into our template(s).
  */
 function draw() {
-  let context = {
-    "subjects": Array.from(subjects.values()),
-    "aspects": Array.from(aspects.values()),
-    "samples": Array.from(samples.values())
+  const subjectList = Array.from(subjects.values());
+  const aspectList = Array.from(aspects.values());
+  const sampleList = Array.from(samples.values())
+  subjectList.forEach(s => s.samples.sort(sampleSorter));
+  aspectList.sort(aspectSorter);
+  const context = {
+    "subjects": subjectList,
+    "aspects": aspectList,
+    "samples": sampleList,
   };
   LENS.innerHTML = mainTemplate(context);
 } // draw
 
+function sampleSorter(a, b) {
```

Comment:
If the rank is the same ret will be set to zero, and then in the final condition it will be sorted by name.